### PR TITLE
Document Update() vs Patch() usage

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

File: developer.md [hunk 1]
```diff
@@ -67,3 +67,23 @@ from its children objects so those can be deleted too.
...
+independent and race-free, an external update to `Spec` can "race" with an
+internal update to `Status` on the generation on the CRD level. This means the
+controller can get a `Conflict` response from k8s when trying to update the
+`Status` of the CR. In our current operator implementations, such conflict
+causes a new reconciliation to start with a fresh copy of the CR. To minimize
+such re-reconciliation attempts the controller should only update the `Status`
+subresource if some status data is changed. The `Helper` in `lib-common`
+tracks the changes on the CR instance and can tell if something is changed. It
+also makes it possible to use `Patch()` over `Update()` to limit the amount of
+data sent to the API server.
```
[the remark this answers is not in the document]
Comment (gibizer) [reply developer.md:89]: Added a paragraph that documents the usage pattern of Helper

File: developer.md [hunk 2]
```diff
@@ -67,3 +67,38 @@ from its children objects so those can be deleted too.
...
+1. At the start of the `Reconcile()` call query the state of the CR from the
+k8s and instantiate a new `Helper` with it. The `Helper` will copy the actual
+state of the CR internally.
+
+2. Modify the CR instance during the `Reconcile()` call
+
+3. At the end of the `Reconcile()` use `Helper.SetAfter()` to provider the new
+state of the CR to the helper. Then use `Helper.GetChanges()` to see if there
+was any changes and if so you can use the controller-runtime's
+`Clien.MergeFrom(Helper.GetBeforeObject())` to generate the patch request that
```
Comment (abays) [inline developer.md:102]: nit: `s/Clien/Client`

Comment (gibizer) [reply developer.md]: Done.

File: developer.md [hunk 3]
```diff
@@ -67,3 +67,38 @@ from its children objects so those can be deleted too.
...
+
+The basic usage pattern of the Helper is the following:
+
+1. At the start of the `Reconcile()` call query the state of the CR from the
+k8s and instantiate a new `Helper` with it. The `Helper` will copy the actual
+state of the CR internally.
+
+2. Modify the CR instance during the `Reconcile()` call
+
+3. At the end of the `Reconcile()` use `Helper.SetAfter()` to provider the new
```
Comment (abays) [inline developer.md:99]: nit: `s/provider/provide`

Comment (gibizer) [reply developer.md]: Done

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**